server/src/test/java/org/elasticsearch/action/bulk/TransportBulkActionIngestTests.java

```diff
@@ -274,7 +276,7 @@ public void testIngestLocal() throws Exception {
         assertFalse(responseCalled.get());
         assertFalse(failureCalled.get());
         verify(ingestService).executeBulkRequest(eq(bulkRequest.numberOfActions()), bulkDocsItr.capture(),
-            failureHandler.capture(), completionHandler.capture(), any());
+            failureHandler.capture(), completionHandler.capture(), any(), eq(Names.WRITE));
```

Comment:
There are a number of instances where we now verify that we're using the `WRITE` threadpool here. Could we add a test here to check the logic that determines when to use `SYSTEM_WRITE`?